linkerd/http/upgrade/src/glue.rs

```diff
@@ -50,9 +51,13 @@ pub struct HyperConnectFuture<F> {
 
 // === impl UpgradeBody ===
 
-impl Body for UpgradeBody {
-    type Data = Bytes;
-    type Error = hyper::Error;
+impl<B> Body for UpgradeBody<B>
+where
+    B: Body + Unpin,
```

Comment:
Correct me if I'm wrong here, but I don't think the `Unpin` bound is necessary here since we mark `B` as `#[pin]` in the type

Reply:
you are right, and i have to commend and thank you for the watchful eye.

in `main` this isn't currently pinned, so this was needed to add the `B` parameter originally in development, but tweaking how we poll the body (_see lines 70-80, below_) means this bound is indeed frivolous now.

039d90bd removes this.